runtime/src/iree-amd-aie/aie_runtime/Utils/ChannelGenerator.h

```diff
@@ -22,45 +22,61 @@ class ChannelGenerator {
   ChannelGenerator() {}
   ChannelGenerator(uint8_t numProducerChannels, uint8_t numConsumerChannels)
       : numProducerChannels(numProducerChannels),
-        numConsumerChannels(numConsumerChannels) {}
+        numConsumerChannels(numConsumerChannels) {
+    assert(numProducerChannels > 0 && numConsumerChannels > 0 &&
+           "Invalid number of producer/consumer channels.");
+    // Initialize to the last channel for round-robin usage.
```

Comment:
I think it makes sense to have an `AssignmentMode` similar to the `ChannelBdIdGenerator`:

```
ChannelAssignmentMode {
  FirstAvailable,
  RoundRobin
}
```

Or if the logic differs quite a bit, use specialized child classes instead: `RoundRobinChannelGenerator`, `FirstAvailableChannelGenerator`.